### PR TITLE
Impl bandersnatch curve

### PR DESCRIPTION
**Pull request by zhenfeizhang**
## Description

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
__See description below.__
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

This PR implements the [Bendersnatch curve](https://ethresear.ch/t/introducing-bandersnatch-a-fast-elliptic-curve-built-over-the-bls12-381-scalar-field/9957) discovered by Simon Masson from [Anoma](https://anoma.network/) and Antonio Sanso from [Ethereum Foundation](https://ethereum.foundation/).

This PR does __NOT__ implement the GLV speed up. This will be on a separate PR.

### CODE REVIEW

**Comment by weikengchen** [comment]
Thanks for this! I saw your Wechat tweet this morning.

**Comment by zhenfeizhang** [comment]
Thanks for the support!

File: ed_on_bls12_381_bandersnatch/src/curves/mod.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#[derive(Clone, Default, PartialEq, Eq)]
+pub struct EdwardsParameters;
+
+impl ModelParameters for EdwardsParameters {
+    type BaseField = Fq;
+    type ScalarField = Fr;
+}
+
+impl TEModelParameters for EdwardsParameters {
+    /// COEFF_A = -1
```
Comment (Pratyush) [inline ed_on_bls12_381_bandersnatch/src/curves/mod.rs:44]: Probably this comment needs to be updated, right?

**Comment by Pratyush** [comment]
Hey @zhenfeizhang, thanks for the PR! Quick question: isn't one of the main benefits of bandersnatch that the Short Weierstrass form has nice GLV properties? So I think it would make sense to add a short weierstrass model (along with the corresponding constraints) here as well.

File: ed_on_bls12_381_bandersnatch/src/curves/mod.rs (hunk 1 above)
Comment (weikengchen) [reply ed_on_bls12_381_bandersnatch/src/curves/mod.rs:44]: Good catch. Noted.

**Comment by zhenfeizhang** [comment]
> Hey @zhenfeizhang, thanks for the PR! Quick question: isn't one of the main benefits of bandersnatch that the Short Weierstrass form has nice GLV properties? So I think it would make sense to add a short weierstrass model (along with the corresponding constraints) here as well.

Good idea. Will look into it.

**Pull request merged**